framework/src/materials/Material.C

```diff
@@ -100,15 +102,25 @@ Material::Material(const InputParameters & parameters) :
 void
 Material::initStatefulProperties(unsigned int n_points)
 {
-  if (_has_stateful_property)
-    for (_qp = 0; _qp < n_points; ++_qp)
-      initQpStatefulProperties();
+  // checking for statefulness of properties via this loop is necessary
+  // because owned props might have been promoted to stateful by calls to
+  // getMaterialProperty[Old/Older] from other objects.  In these cases, this
+  // object won't otherwise know that it owns stateful properties.
+  for (auto& prop : _supplied_props)
```

Comment:
I do not think we want to change this function as we discussed. Normal materials does not need this and it could create the conflict between 'initial condition' and 'initial evaluation'.